ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/Situation.java

```diff
@@ -0,0 +1,299 @@
+            }
+        }
+    }
+
+    /**
+     * This method works from the working variant. It is applying on the network working variant
+     * according to the values present in the CRAC result extension of the working variant.
+     */
+    public void applyRangeActionResultsOnNetwork() {
+        if (workingVariantId == null) {
+            throw new FaraoException(NO_WORKING_VARIANT);
+        }
+        String preventiveState = getCrac().getPreventiveState().getId();
+        for (RangeAction rangeAction : getCrac().getRangeActions()) {
+            RangeActionResultExtension rangeActionResultMap = rangeAction.getExtension(RangeActionResultExtension.class);
+            rangeAction.apply(getNetwork(), rangeActionResultMap.getVariant(workingVariantId).getSetPoint(preventiveState));
+        }
+    }
+
+    /**
+     * This method compares CRAC result extension of two different variants. It compares the set point values
+     * of all the range actions.
+     *
+     * @param variantId1: First variant to compare.
+     * @param variantId2: Second variant to compare.
+     * @return True if all the range actions are set at the same values and false otherwise.
+     */
+    public boolean sameRemedialActions(String variantId1, String variantId2) {
+        String preventiveState = getCrac().getPreventiveState().getId();
+        for (RangeAction rangeAction : getCrac().getRangeActions()) {
+            RangeActionResultExtension rangeActionResultMap = rangeAction.getExtension(RangeActionResultExtension.class);
+            double value1 = rangeActionResultMap.getVariant(variantId1).getSetPoint(preventiveState);
+            double value2 = rangeActionResultMap.getVariant(variantId2).getSetPoint(preventiveState);
+            if (value1 != value2 && (!Double.isNaN(value1) || !Double.isNaN(value2))) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    // VARIANTS MANAGEMENT METHODS
+
+    /**
+     * This method generates a unique variant ID of a network according to the IDs already present in the network.
+     *
+     * @return A unique ID of the network as a string.
+     */
+    private String getUniqueNetworkVariantId() {
+        String uniqueId;
+        do {
+            uniqueId = UUID.randomUUID().toString();
+        } while (network.getVariantManager().getVariantIds().contains(uniqueId));
+        return uniqueId;
+    }
+
+    /**
+     * This method generates a unique variant ID of a situation. This ID has to be unique for the network
+     * and for the CRAC.
+     *
+     * @return A unique ID a the situation as a string.
+     */
+    private String getUniqueSituationId() {
+        String networkVariantId = getUniqueNetworkVariantId();
+        String cracVariantId = crac.getExtension(ResultVariantManager.class).getUniqueVariantId();
+        return networkVariantId.concat(cracVariantId);
+    }
```

Comment:
cannot we simplify in only one method getUniqueVariantId() which generated new unique ID and which verifies that it doesn't exists in 
- the network
- the crac
- the variantIds

Reply:
and this method could take a prefix